libs/filamat/src/shaders/CodeGenerator.cpp

```diff
@@ -460,19 +491,22 @@ io::sstream& CodeGenerator::generateOutput(io::sstream& out, ShaderStage type,
     const char* materialTypeString = getOutputTypeName(materialOutputType);
     const char* typeString = getOutputTypeName(outputType);
 
+    bool generate_essl3_code = mTargetLanguage == TargetLanguage::SPIRV
```

Comment:
why can't we generate essl1 with target lang spirv?

Reply:
I described it a bit above in a comment elsewhere in this file:

    // During compilation and optimization, __VERSION__ reflects the shader language version of the
    // intermediate code, not the version of the final code. spirv-cross automatically adapts
    // certain language features (e.g. fragment output) but leaves others untouched (e.g. sampler
    // functions, bit shift operations). Client code may have to make decisions based on this
    // information, so define a FILAMENT_EFFECTIVE_VERSION constant.

Since the intermediate code is always ESSL 3.0 even if it eventually becomes ESSL 1.0, we have to sometimes generate ESSL 3.0 stuff like this.